### PR TITLE
Adds reinforced rock for mapping use

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

adds reinforced rocks, they are rocks with metal around them to act as a regular wall, when the wall cover is broken it creates a rock

## Why It's Good For The Game

it can be used to give some nice atmosphere, like in this example photo from vg (ingame it smooths)
image

## Changelog
:cl: Fikou, sprites from /vg/
expansion: reinforced rock for mapping use
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
would be nice if it were craftable ingame by slapping asteroid walls with metal, otherwise I feel like we shouldn't add this until a map actually needs it. unused bloat and all

File: code/game/turfs/closed/wall/misc_walls.dm [hunk 1]
```diff
@@ -110,3 +110,19 @@
 	sheet_type = /obj/item/stack/sheet/bronze
 	sheet_amount = 2
 	girder_type = /obj/structure/girder/bronze
+
+/turf/closed/wall/rock
+	name = "reinforced rock"
+	desc = "It has metal struts that need to be welded away before it can be mined."
```
Comment (MrDoomBringer) [inline code/game/turfs/closed/wall/misc_walls.dm:116]: Feel like this should use the whole deconstruction description-tips we have for reinforced walls
ie
desc = "A rock wall reinforced by a metal support structure."
"The struts are <b>welded</b> together at the seams"

**Comment by maxymax13** [comment]
make it upgradable from rocks ingame so you can make your tram/kilo crack shacks if you have some iron without having to do the *whole* wall making process just to make your walls not crumble if an assistant breathes on them

**Review by EOBGames — APPROVED** [review]
I'd like to have this, especially if you can add other variants for other types of rock, like the type used on icebox.

**Review by Maurukas — APPROVED** [review]
More variety in appearance for asteroids is good.

File: code/game/turfs/closed/wall/misc_walls.dm (hunk 1 above)
Comment (coiax) [reply code/game/turfs/closed/wall/misc_walls.dm:116]: Won't it already have the deconstruction tooltips by being a wall, or nah?

File: code/game/turfs/closed/walls.dm [hunk 2]
```diff
@@ -27,6 +27,8 @@
 	var/sheet_type = /obj/item/stack/sheet/iron
 	var/sheet_amount = 2
 	var/girder_type = /obj/structure/girder
+	/// A turf that will replace this turf when this turf is destroyed
+	var/decon_type
```
Comment (coiax) [inline code/game/turfs/closed/walls.dm:31]: Do we need both a girder and a decon type? Could these two be combined?

File: code/game/turfs/closed/minerals.dm [hunk 3]
```diff
@@ -517,6 +517,11 @@
 	smooth_icon = 'icons/turf/walls/red_wall.dmi'
 	base_icon_state = "red_wall"
 
+/turf/closed/mineral/asteroid/porous
+	name = "porous rock"
+	desc = "This rock is filled with pockets of breathable air."
+	baseturfs = /turf/open/floor/plating/asteroid
```
Comment (coiax) [inline code/game/turfs/closed/minerals.dm:523]: So basically, this is rock when mined, makes red floor full of air?

**Review by Twaticus — APPROVED** [review]

File: code/game/turfs/closed/minerals.dm (hunk 3 above)
Comment (Fikou) [reply code/game/turfs/closed/minerals.dm:523]: yeah, so like if you put these in maint, you dont want 0 pressure if its mined, youd like it to have a bit of air.

File: code/game/turfs/closed/walls.dm (hunk 2 above)
Comment (Fikou) [reply code/game/turfs/closed/walls.dm:31]: id have to do some weird isturf() checks probably, i can do it if you insist though

File: code/game/turfs/closed/wall/misc_walls.dm (hunk 1 above)
Comment (Fikou) [reply code/game/turfs/closed/wall/misc_walls.dm:116]: its inherited from parent

**Review by coiax — CHANGES_REQUESTED** [review]
So as me and @ninjanomnom spoke about in Discord, the metal-encased-asteroid rock should be a structure that's on the asteroid turf, that needs to be dismantled before you're allowed to mine it.

**Comment by Fikou** [comment]
> So as me and @ninjanomnom spoke about in Discord, the metal-encased-asteroid rock should be a structure that's on the asteroid turf, that needs to be dismantled before you're allowed to mine it.

it shouldnt

**Comment by MrDoomBringer** [comment]
Why not make this a component that can be applied to all turfs that just gives them the metal overlay and makes them unminable/gives higher durability?

**Comment by Fikou** [comment]
h

**Comment by Fikou** [comment]
> So as me and @ninjanomnom spoke about in Discord, the metal-encased-asteroid rock should be a structure that's on the asteroid turf, that needs to be dismantled before you're allowed to mine it.

i have tested a structure, even ignoring the fact that i'd need to make new sprites, it doesnt work with how mining rocks works. mining drills go through, projectiles go through, i need to copy random wall code to make it be similar, this isnt viable

**Comment by Fikou** [comment]
@coiax

**Comment by AMonkeyThatCodes** [comment]
1. self assign pr
2. never review it
3. refuse to elaborate further

**Comment by coiax** [comment]
Leaving this open for someone else.

**Comment by Mothblocks** [comment]
i think its fine as is and i dont really want it sitting around with the 110 other prs up right now

**Pull request merged**